samples/apps/copilot-chat-app/webapi/Controllers/SemanticKernelController.cs

```diff
@@ -42,7 +46,11 @@ public SemanticKernelController(IServiceProvider serviceProvider, IConfiguration
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
-    public async Task<ActionResult<AskResult>> InvokeFunctionAsync([FromServices] Kernel kernel, [FromBody] Ask ask,
+    public async Task<ActionResult<AskResult>> InvokeFunctionAsync(
+        [FromServices] Kernel kernel,
+        [FromServices] ChatSessionRepository chatRepository,
```

Comment:
I'm not a big fan of making these dependencies a part of the generic call.  I guess because of the time pressure, we can do things this way now but eventually, we'll need to figure out a way to make things independent

Reply:
It's a very simple change to move those dependencies to constructor of this controller.

Reply:
Either way there is going to be a coupling between the backend service and copilot chat (for now), right?

Reply:
Yes, for now at least.

But using dependency at the constructor level is a step in the right direction.

Reply:
Agreed!